org.metaborg.spoofax.shell.core/src/main/java/org/metaborg/spoofax/shell/core/ClassPathInterpreterLoader.java

```diff
@@ -0,0 +1,148 @@
+
+        String mimeType = entryPoint.getMimeType();
+        PolyglotEngine builtEngine =
+            PolyglotEngine.newBuilder().config(mimeType, DynSemLanguage.PARSER, parser)
+                .config(mimeType, DynSemLanguage.RULE_REGISTRY, ruleRegistry)
+                .config(mimeType, DynSemLanguage.TERM_REGISTRY, termRegistry).build();
+        try {
+            InputStreamReader specTermReader =
+                new InputStreamReader(entryPoint.getSpecificationTerm(), "UTF-8");
+            builtEngine.eval(Source.fromReader(specTermReader, "Evaluate to interpreter.")
+                .withMimeType(mimeType));
+        } catch (IOException e) {
+            throw new InterpreterLoadException(e);
+        }
+        return builtEngine;
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public ITerm getProgramTerm(IStrategoTerm input) throws InterpreterLoadException {
+        try {
+            String termSort = StrategoUtil.getSortForTerm(input);
+            // Get the abstract class for the sort of the term.
+            Class<? extends ITerm> generatedTermClass = (Class<? extends ITerm>) ClassUtils
+                .getClass(targetPackage + ".terms.I" + termSort + "Term");
+            return (ITerm) MethodUtils.invokeStaticMethod(generatedTermClass, "create", input);
+        } catch (ClassNotFoundException | NoSuchMethodException | IllegalAccessException
+                 | InvocationTargetException cause) {
+            throw new InterpreterLoadException("Error constructing program term from input.",
+                                               cause);
+        }
+    }
+
+    @Override
+    public ITermTransformer getTransformer() {
+        return transformer;
+    }
+
+    private DynSemEntryPoint getEntryPoint() throws InterpreterLoadException {
+        try {
+            Class<DynSemEntryPoint> entryPointClass =
+                this.<DynSemEntryPoint> getGeneratedClass("EntryPoint");
+            return ConstructorUtils.invokeConstructor(entryPointClass);
+        } catch (InstantiationException | IllegalAccessException | NoSuchMethodException
+                 | InvocationTargetException e) {
+            throw new InterpreterLoadException(e);
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    private <T> Class<T> getGeneratedClass(String className) throws InterpreterLoadException {
+        try {
+            return (Class<T>) ClassUtils.getClass(targetPackage + "." + langName + className);
+        } catch (ClassNotFoundException e) {
+            throw new InterpreterLoadException(e);
+        }
+    }
+
+    /* Loads the required configurations from the dynsem.properties file parsed as a Properties
+     * object. */
+    private void loadDynSemProperties(ILanguageImpl langImpl) throws InterpreterLoadException {
+        FileObject dynSemPropertiesFile = findDynSemPropertiesFileForLanguage(langImpl);
+
+        Properties dynSemProperties = new Properties();
+        InputStream in;
```

Comment:
1. This can go inside the `try` (it's not used after the `try/catch` block.
2. Shouldn't this `InputStream` be closed?